### ISSUE TITLE
ExceptionGroup module not found

### ISSUE THREAD

**Issue by dev4s** [Garulf/Flow-Raindrop#4]
Hi,
when trying to run the plugin it shows that the module Exception group is not found. Creating folder exceptiongroup and downloading the module from pypi helped with running this plugin.

Best regards,

**Comment by Garulf** [comment]
That's... strange... do you have the error log for this?

**Comment by Gowwre** [comment]
I have also encountered this error when trying to run the plugin. Here's the error:

```
Flow Launcher version: 1.18.0


OS Version: 22631.2715


IntPtr Length: 8
x64: True

Python Path: D:\Python\pythonw.exe
Node Path: 
Date: 06/08/2024 10:39:32
Exception:
FlowRaindrop Exception: 
Websites: https://github.com/Garulf/flow-raindrop
Author: Garulf
Version: 1.0.0
Flow.Launcher.Core.ExternalPlugins.FlowPluginException: Traceback (most recent call last):
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\run.py", line 11, in <module>
    from plugin.main import plugin
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\plugin\main.py", line 6, in <module>
    from raindrop import Raindrop
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\plugin\raindrop.py", line 4, in <module>
    import hishel
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\__init__.py", line 3, in <module>
    from ._async import *
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_async\__init__.py", line 1, in <module>
    from ._client import *  # noqa: F403
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_async\_client.py", line 5, in <module>
    from hishel._async._transports import AsyncCacheTransport
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_async\_transports.py", line 10, in <module>
    from hishel._utils import extract_header_values_decoded, normalized_url
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_utils.py", line 7, in <module>
    import anyio
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\anyio\__init__.py", line 27, in <module>
    from ._core._sockets import connect_tcp as connect_tcp
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\anyio\_core\_sockets.py", line 35, in <module>
    from exceptiongroup import ExceptionGroup
ModuleNotFoundError: No module named 'exceptiongroup'

 ---> System.IO.InvalidDataException: Traceback (most recent call last):
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\run.py", line 11, in <module>
    from plugin.main import plugin
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\plugin\main.py", line 6, in <module>
    from raindrop import Raindrop
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\plugin\raindrop.py", line 4, in <module>
    import hishel
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\__init__.py", line 3, in <module>
    from ._async import *
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_async\__init__.py", line 1, in <module>
    from ._client import *  # noqa: F403
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_async\_client.py", line 5, in <module>
    from hishel._async._transports import AsyncCacheTransport
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_async\_transports.py", line 10, in <module>
    from hishel._utils import extract_header_values_decoded, normalized_url
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\hishel\_utils.py", line 7, in <module>
    import anyio
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\anyio\__init__.py", line 27, in <module>
    from ._core._sockets import connect_tcp as connect_tcp
  File "C:\Users\Radriar\AppData\Roaming\FlowLauncher\Plugins\FlowRaindrop-1.0.0\lib\anyio\_core\_sockets.py", line 35, in <module>
    from exceptiongroup import ExceptionGroup
ModuleNotFoundError: No module named 'exceptiongroup'

   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 243
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 248
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.QueryAsync(Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 261
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass23_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 196
--- End of stack trace from previous location ---
   at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 40
   at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 230
   --- End of inner exception stack trace ---
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass23_1.<QueryForPluginAsync>b__1(ActionContext _) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 224
   at Flow.Launcher.Plugin.Result.ExecuteAsync(ActionContext context) in /_/Flow.Launcher.Plugin/Result.cs:line 246
   at Flow.Launcher.ViewModel.MainViewModel.OpenResultAsync(String index) in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 307
   at CommunityToolkit.Mvvm.Input.AsyncRelayCommand.AwaitAndThrowIfFailed(Task executionTask)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

[2024-06-08.txt](https://github.com/user-attachments/files/15746351/2024-06-08.txt)

**Comment by Garulf** [comment]
Looks like one of the libraries I am using is using a Python 3.11 only feature. 

This can be easily resolved by using Python 3.11 or greater. Please update your Python.

**Comment by Gowwre** [comment]
I have updated to the latest Python version and it works perfectly. Thanks a lot!
